core/src/main/scala/cats/data/IdT.scala

```diff
@@ -27,24 +27,86 @@ package data
  */
 final case class IdT[F[_], A](value: F[A]) {
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.data.IdT
+   *
+   *  scala> val idT: IdT[List, Int] = IdT(List(1, 2, 3))
+   *  scala> idT.map(_ + 1)
+   *  res0: IdT[List, Int] = IdT(List(2, 3, 4))
+   * }}}
+   */
   def map[B](f: A => B)(implicit F: Functor[F]): IdT[F, B] =
     IdT(F.map(value)(f))
 
   /**
    * Modify the context `F` using transformation `f`.
+   *
+   * Example:
+   * {{{
+   *  scala> import cats.~>
+   *  scala> import cats.data.IdT
+   *
+   *  scala> val vectorToList: Vector ~> List = new ~>[Vector, List] { override def apply[A](v: Vector[A]): List[A] = v.toList }
```

Comment:
This part can be simplified :
```suggestion
   *  scala> val vectorToList: Vector ~> List = FunctionK.liftFunction(_.toList)
```
(requires `import cats.arrow.FunctionK`)

Reply:
(for Scala3 the syntax is going to be slightly different though)

Reply:
Well, I'm leaning to think that examples in scaladocs should keep the examples as simple as possible. Whereas the details on how, let's say, `FunctionK` works is supposed to be captured in the `FunctionK` scaladocs per se.

But I don't think it's really worth bikeshedding anyway – in any case it is a great job on providing the missing docs for the class, thank you for doing this!

Reply:
I'm not sure if we can write it like that, don't the doctests have to cross-compile for Scala 2 and Scala 3? So we are stuck with worst-of-both-worlds syntax.